### ISSUE TITLE
Move @PicocliScript and Groovy dependenies to separate artifact

### ISSUE THREAD

**Issue by talios** [remkop/picocli#479]
Is it possible to extract/move:

```
picocli/groovy/
picocli/groovy/PicocliBaseScript.class
picocli/groovy/PicocliScript.class
picocli/groovy/PicocliScriptASTTransformation.class
```
to a separate JAR artifact, ideally I don't want to pull in Groovy into my app since I don't need it, nor does picocli itself.

**Comment by talios** [comment]
In this instance I was experimenting with trying to generate a jlink image under JDK11ea, and using a maven plugin to generate a modular version of the jar, which picks up the classes inside the jar.

All good and not a major issue - I suspect we'll find a lot of library projects _may_ start hitting similar issues ( unfortunately, I just discovered Automatic Modules can't be used with jlink ).

**Comment by remkop** [comment]
@talios, FYI it _is_ possible to use jlink with non-modular applications. Simon Ritter wrote this excellent tutorial: https://medium.com/azulsystems/using-jlink-to-build-java-runtimes-for-non-modular-applications-9568c5e70ef4

**Comment by remkop** [comment]
Thinking to address this in picocli 4.0 as detailed here: https://github.com/remkop/picocli/issues/495#issuecomment-437568224

**Comment by remkop** [comment]
FYI: #495 has just landed in master and will be included in the upcoming 4.0.0-alpha-2 release.
Closing this ticket. Feedback very welcome!
